templates/lb-fargate-service/cf.yml

```diff
@@ -247,4 +291,48 @@ Resources:
       ListenerArn:
         Fn::ImportValue:
           !Sub "${ProjectName}-${EnvName}-HTTPSListenerArn"
-      Priority: !Ref RulePriority
+      Priority: !GetAtt HTTPSRulePriorityAction.Priority
+
+  HTTPRulePriorityAction:
+    Condition: HTTPLoadBalancer
+    Type: Custom::RulePriorityFunction
+    Properties:
+      ServiceToken: !GetAtt RulePriorityFunction.Arn
+      ListenerArn:
+        Fn::ImportValue:
+          !Sub "${ProjectName}-${EnvName}-HTTPListenerArn"
+
+  HTTPListenerRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Condition: HTTPLoadBalancer
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref TargetGroup
+          Type: forward
+      Conditions:
+        - Field: 'path-pattern'
+          PathPatternConfig:
+            Values:
+            - !Ref RulePath
+      ListenerArn:
+        Fn::ImportValue:
+          !Sub "${ProjectName}-${EnvName}-HTTPListenerArn"
+      Priority: !GetAtt HTTPRulePriorityAction.Priority
+
+  # Force a conditional dependency from the ECS service on the listener rules.
+  HTTPSWaitHandle:
+    Condition: HTTPSLoadBalancer
+    DependsOn: HTTPSListenerRule
+    Type: "AWS::CloudFormation::WaitConditionHandle"
+
+  HTTPWaitHandle:
+    Condition: HTTPLoadBalancer
+    DependsOn: HTTPListenerRule
+    Type: "AWS::CloudFormation::WaitConditionHandle"
+
+  WaitCondition:
```

Comment:
Should we rename to `WaitUntilListenerRuleIsCreated`?

Reply:
Sure!